legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/auth/ServiceStoreConnectionProvider.java

```diff
@@ -0,0 +1,105 @@
+
+package org.finos.legend.engine.plan.execution.stores.service.auth;
+
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.eclipse.collections.impl.utility.ListIterate;
+import org.finos.legend.authentication.credentialprovider.CredentialProviderProvider;
+import org.finos.legend.engine.connection.ConnectionProvider;
+import org.finos.legend.engine.connection.ConnectionSpecification;
+import org.finos.legend.engine.plan.execution.stores.service.IServiceStoreExecutionExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.AuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.SingleAuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.specification.AuthenticationSpecification;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.service.model.SecurityScheme;
+import org.finos.legend.engine.shared.core.function.Function5;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ServiceStoreConnectionProvider extends ConnectionProvider<HttpConnectionBuilder>
+{
+    public ServiceStoreConnectionProvider(CredentialProviderProvider credentialProviderProvider)
+    {
+        super(credentialProviderProvider);
+    }
+
+    public HttpConnectionBuilder makeConnection(ConnectionSpecification connectionSpecification, AuthenticationSpecification authenticationSpecification, Identity identity) throws Exception
+    {
+        if (!(connectionSpecification instanceof ServiceStoreConnectionSpecification && authenticationSpecification instanceof ServiceStoreAuthenticationSpecification))
+        {
+            throw new IllegalStateException("Invalid ConnectionSpecification/AuthenticationSpecification. Please reach out to dev team");
+        }
+
+        ServiceStoreConnectionSpecification serviceStoreConnectionSpecification = (ServiceStoreConnectionSpecification) connectionSpecification;
+        ServiceStoreAuthenticationSpecification serviceStoreAuthenticationSpecification = (ServiceStoreAuthenticationSpecification) authenticationSpecification;
+
+        for (AuthenticationSchemeRequirement requirement : serviceStoreAuthenticationSpecification.authenticationSchemeRequirements)
+        {
+            SingleAuthenticationSchemeRequirement authenticationSchemeRequirement = (SingleAuthenticationSchemeRequirement) requirement;
+            SecurityScheme securityScheme = authenticationSchemeRequirement.securityScheme;
+            AuthenticationSpecification authSpecification = authenticationSchemeRequirement.authenticationSpecification;
+
+            HttpClientBuilder clientBuilder = HttpClients.custom();
```

Comment:
should next 3 lines be outside loop, it seems we over-writing the updates, right?

Reply:
We cant move them outside. So, lets say we have multiple authRequirements. 
First pass, we go and fetch cred1 for scheme1 and update requestBuilder/clientbuilder with cred1. But when we try to make connection, request fails. In that case, we try to get cred2 for scheme2 and we will need fresh copies of clinetBuilder and requestBuilder to which we can add cred. 

We dont have a check right now to see if authentication fails for first pass and then only process other security schemes. But going forward, we will need some sort of check for multiple security schemes/authRequirements.

Reply:
lets add try/catch here